Allow images without separate /efi & /boot partitions

valentindavid opened the pull request:
This is for GNOME OS where we use systemd boot where we do not require
the use of a separate partition.

https://phabricator.endlessm.com/T30030

valentindavid committed:
Allow images without separate /efi & /boot partitions

This is for GNOME OS where we use systemd boot where we do not require
the use of a separate partition.

eos-shell-build commented:
I don't recognize the submitter as a member of @endlessm. Can somebody in @endlessm verify the patch? See https://phabricator.endlessm.com/w/software/build/ci-integration/#pull-requests for more details.

wjt commented:
ok to test

Thanks for this – sorry in advance for a slightly delayed review

valentindavid commented:
@wjt The link for the details of the build does not seem to work. `make distcheck` seems to work for me. Could you provide information on what check failed?

wjt commented:
Sorry, that build server is not public.

```
gpt.c: In function ‘is_eos_gpt_valid’:
gpt.c:204:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
     struct gpt_header testcrc_header;
     ^~~~~~
gpt.c:213:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
     int n = pt->header.ptable_count * pt->header.ptable_partition_size;
     ^~~
gpt.c:231:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
     int has_root = 0;
     ^~~
gpt.c:232:19: error: comparison of integer expressions of different signedness: ‘int’ and ‘uint32_t’ {aka ‘unsigned int’} [-Werror=sign-compare]
     for (i = 1; i < pt->header.ptable_count; ++i) {
                   ^
gpt.c: In function ‘get_is_valid_eos_gpt’:
gpt.c:262:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
     FILE *in_file = fopen(filepath, "r");
     ^~~~
gpt.c:264:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
     struct ptable pt;
     ^~~~~~
At top level:
gpt.c:6:16: error: ‘GPT_GUID_BOOT’ defined but not used [-Werror=unused-variable]
 static uint8_t GPT_GUID_BOOT[] = {0x48, 0x61, 0x68, 0x21, 0x49, 0x64, 0x6f, 0x6e, 0x74, 0x4e, 0x65, 0x65, 0x64, 0x45, 0x46, 0x49};
                ^~~~~~~~~~~~~
```

appears to be the issue.

valentindavid commented:
@wjt Sorry for the delay. Now it should build without warning.

wjt commented:
Thanks, I pushed a few comment tweaks, and tested this locally. It still successfully detected an Endless OS image as expected.

> @wjt Sorry for the delay.

All I can say is, please forgive my subsequent 11 month delay.

wjt committed:
Remove intltool lock file on distclean

This is a workaround for
https://bugs.launchpad.net/intltool/+bug/1712194:

    ERROR: files left in build directory after distclean:
    ./po/.intltool-merge-cache.lock
